b2sdk/replication/setting.py

```diff
@@ -191,65 +157,54 @@ def as_dict(self) -> dict:
+                } if self.is_source else None,
+            'asReplicationDestination':
+                {
+                    'sourceToDestinationKeyMapping': self.source_to_destination_key_mapping,
+                } if self.is_destination else None,
+        }
+
         return result
 
+    serialize_to_json_for_request = as_dict
+
     @classmethod
     def from_dict(cls, value_dict: dict) -> 'ReplicationConfiguration':
-        replication_source_dict = value_dict.get('asReplicationSource')
-        as_replication_source = ReplicationSourceConfiguration.from_dict(
-            replication_source_dict
-        ) if replication_source_dict else ReplicationSourceConfiguration()
-
-        replication_destination_dict = value_dict.get('asReplicationDestination')
-        as_replication_destination = ReplicationDestinationConfiguration.from_dict(
-            replication_destination_dict
-        ) if replication_destination_dict else ReplicationDestinationConfiguration()
+        source_dict = value_dict.get('asReplicationSource') or {}
+        destination_dict = value_dict.get('asReplicationDestination') or {}
 
         return cls(
-            as_replication_source=as_replication_source,
-            as_replication_destination=as_replication_destination,
+            rules=[
+                ReplicationRule.from_dict(rule_dict)
+                for rule_dict in source_dict.get('replicationRules', [])
+            ],
+            source_key_id=source_dict.get('sourceApplicationKeyId'),
+            source_to_destination_key_mapping=destination_dict.get('sourceToDestinationKeyMapping')
+            or {},
         )
 
 
-@dataclass
+@dataclass(kw_only=True, slots=True)
 class ReplicationConfigurationFactory:
     is_client_authorized_to_read: bool
     value: Optional[ReplicationConfiguration]
 
     @classmethod
-    def from_bucket_dict(cls, bucket_dict: dict) -> Optional['ReplicationConfigurationFactory']:
+    def from_bucket_dict(cls, bucket_dict: dict) -> 'ReplicationConfigurationFactory':
         """
         Returns ReplicationConfigurationFactory for the given bucket dict
-        retrieved from the api, or None if no replication configured.
+        retrieved from the api.
         """
-        replication_dict = bucket_dict.get('replicationConfiguration')
-        if not replication_dict:
-            return cls(
-                is_client_authorized_to_read=True,
-                value=ReplicationConfiguration(),
-            )
-
-        return cls.from_dict(replication_dict)
-
-    @classmethod
-    def from_dict(cls, value_dict: dict) -> 'ReplicationConfigurationFactory':
-        if not value_dict['isClientAuthorizedToRead']:
-            return cls(
-                is_client_authorized_to_read=False,
-                value=None,
-            )
+        replication_dict = bucket_dict.get('replicationConfiguration') or {}
+        value_dict = replication_dict.get('value') or {}
```

Comment:
```suggestion
        replication_dict = bucket_dict.get('replicationConfiguration', {})
        value_dict = replication_dict.get('value', {})
```

Reply:
Original variant covers cases when `bucket_dict['replicationConfiguration']` is present but is `None` for some reason - even in such cases we get `{}`.